src/main/java/frc/robot/subsystems/swerve/SwerveSubsystem.java

```diff
@@ -23,40 +23,60 @@
 
-public class Swerve extends SubsystemBase {
-  private static final double MAX_LINEAR_SPEED = Units.feetToMeters(14.5);
-  private static final double TRACK_WIDTH_X = Units.inchesToMeters(25.0);
-  private static final double TRACK_WIDTH_Y = Units.inchesToMeters(25.0);
-  private static final double DRIVE_BASE_RADIUS =
+public class SwerveSubsystem extends SubsystemBase {
+  // Drivebase constants
+  public static final double MAX_LINEAR_SPEED = Units.feetToMeters(14.5);
+  public static final double TRACK_WIDTH_X = Units.inchesToMeters(25.0);
+  public static final double TRACK_WIDTH_Y = Units.inchesToMeters(25.0);
+  public static final double DRIVE_BASE_RADIUS =
       Math.hypot(TRACK_WIDTH_X / 2.0, TRACK_WIDTH_Y / 2.0);
-  private static final double MAX_ANGULAR_SPEED = MAX_LINEAR_SPEED / DRIVE_BASE_RADIUS;
+  public static final double MAX_ANGULAR_SPEED = MAX_LINEAR_SPEED / DRIVE_BASE_RADIUS;
+  // Hardware constants
+  public static final int PigeonID = 0;
+  // FL
+  public static final int flDriveID = 0;
+  public static final int flTurnID = 1;
+  public static final int flCancoderID = 0;
+  public static final Rotation2d flCancoderOffset = Rotation2d.fromRotations(0.0);
+  // FR
+  public static final int frDriveID = 2;
+  public static final int frTurnID = 3;
+  public static final int frCancoderID = 1;
+  public static final Rotation2d frCancoderOffset = Rotation2d.fromRotations(0.0);
+  // BL
+  public static final int blDriveID = 4;
+  public static final int blTurnID = 5;
+  public static final int blCancoderID = 2;
+  public static final Rotation2d blCancoderOffset = Rotation2d.fromRotations(0.0);
+  // BR
+  public static final int brDriveID = 6;
+  public static final int brTurnID = 7;
+  public static final int brCancoderID = 3;
+  public static final Rotation2d brCancoderOffset = Rotation2d.fromRotations(0.0);
 
   public static final Lock odometryLock = new ReentrantLock();
   private final GyroIO gyroIO;
   private final GyroIOInputsAutoLogged gyroInputs = new GyroIOInputsAutoLogged();
-  private final Module[] modules = new Module[4]; // FL, FR, BL, BR
+  private final Module[] modules; // FL, FR, BL, BR
 
   private SwerveDriveKinematics kinematics = new SwerveDriveKinematics(getModuleTranslations());
   private Pose2d pose = new Pose2d();
   private Rotation2d lastGyroRotation = new Rotation2d();
 
-  public Swerve(
-      GyroIO gyroIO,
-      ModuleIO flModuleIO,
-      ModuleIO frModuleIO,
-      ModuleIO blModuleIO,
-      ModuleIO brModuleIO) {
+  public SwerveSubsystem(GyroIO gyroIO, ModuleIO... moduleIOs) {
     this.gyroIO = gyroIO;
-    modules[0] = new Module(flModuleIO, 0);
-    modules[1] = new Module(frModuleIO, 1);
-    modules[2] = new Module(blModuleIO, 2);
-    modules[3] = new Module(brModuleIO, 3);
+    modules = new Module[moduleIOs.length];
+    for (int i = 0; i < moduleIOs.length; i++) {
+      modules[i] = new Module(moduleIOs[i], Integer.toString(i));
```

Comment:
If you're going to make the second arg a suffix instead of an id, maybe you want to use something more semantically useful like "front left" etc?